src/globus_cli/commands/cli_profile_list.py

```diff
@@ -0,0 +1,97 @@
+import os
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import click
+import globus_sdk.tokenstorage
+
+from globus_cli.login_manager import token_storage_adapter
+from globus_cli.parsing import command
+from globus_cli.termio import FORMAT_TEXT_TABLE, formatted_print
+from globus_cli.types import FIELD_LIST_T
+
+
+# TODO: upstream this into the SDK as a method of the SQLiteStorageAdapter
+def _iter_namespaces(adapter: globus_sdk.tokenstorage.SQLiteAdapter) -> Iterator[str]:
+    conn = adapter._connection
+
+    cursor = conn.execute("SELECT DISTINCT namespace FROM token_storage;")
+    for row in cursor:
+        yield row[0]
+
+
+def _profilestr_to_datadict(s: str) -> Optional[Dict[str, Any]]:
+    if s.count("/") < 1:
+        return None
+    if s.count("/") < 2:
+        category, env = s.split("/")
+        if category == "clientprofile":  # should not be possible
+            return None
```

Comment:
Should we print a warning or something if we do somehow hit a "shouldn't be possible" case?

Reply:
I was avoiding doing so because the only way I can imagine these cases existing is someone tinkering with their DB directly. I think that at least by default we should ignore anything unrecognized.
Maybe with `--all` we could provide a list of unrecognized namespaces?

Reply:
Probably fine to leave this as is then

Reply:
If we find that we want this at some point, we can always add an `if all: warnecho(...)` here. But yeah, let's leave it out for now.